web/src/sagas/index.js

```diff
@@ -0,0 +1,85 @@
+import {
+  call,
+  put,
+  select,
+  takeLatest,
+} from 'redux-saga/effects';
+
+import thirdPartyServices from '../services/thirdparty';
+import { handleConnectionReturnCode, protectedJsonRequest, textRequest } from '../helpers/api';
+import {
+  getGfsTargetTimeBefore,
+  getGfsTargetTimeAfter,
+  fetchGfsForecast,
+} from '../helpers/gfs';
+
+function* fetchClientVersion(action) {
+  try {
+    const version = yield call(textRequest, '/clientVersion');
+    yield put({ type: 'APPLICATION_STATE_UPDATE', key: 'version', value: version });
+  } catch (error) {
+    const appState = yield select(state => state.application);
+    handleConnectionReturnCode(error, appState);
+  }
+}
+
+function* fetchZoneHistory(action) {
+  const { zoneId } = action.payload;
+  try {
+    const payload = yield call(protectedJsonRequest, `/v3/history?countryCode=${zoneId}`);
+    yield put({ type: 'ZONE_HISTORY_FETCH_SUCCEEDED', zoneId, payload });
+  } catch (error) {
+    yield put({ type: 'ZONE_HISTORY_FETCH_FAILED', error });
+  }
+}
+
+function* fetchGridData(action) {
+  const { datetime } = action.payload;
+  try {
+    const payload = yield call(protectedJsonRequest, datetime ? `/v3/state?datetime=${datetime}` : '/v3/state');
+    thirdPartyServices.trackWithCurrentApplicationState('pageview');
```

Comment:
@corradio I kept this line here since it was there before, but I'm not sure it really belongs here - was the intention to track page views on every grid data fetch? We already do it on every route change - [see here](https://github.com/tmrowco/electricitymap-contrib/pull/2388/files/637bc3a7888cce7bdd9d9f070aceda52de314fed#diff-42c3276bc26032ecad9d7061dd401cd4R474).

Reply:
yes I think the purpose was to track a pageview when the data refreshes (some people have TVs which are left running with emap)

Reply:
Ok thanks! Maybe it's the name `pageview` that confused me as it immediatelly made me assume the event is triggered by a user action. Maybe it's something worth discussing in general, but I won't touch it here then! :)